spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/AsyncComponentAdapters.java

```diff
@@ -214,6 +229,42 @@ public CompletableFuture<Void> onMessage(Collection<Message<T>> messages) {
 		}
 	}
 
+	private static class FilteredMessageListenerAdapter<T> extends AbstractThreadingComponentAdapter
+		implements AsyncMessageListener<T> {
+
+		private final MessageListener<T> filteredMessageListener;
+		private final MessageFilter<T> filter;
+
+		public FilteredMessageListenerAdapter(MessageListener<T> filteredMessageListener, MessageFilter<T> filter) {
+			this.filteredMessageListener = filteredMessageListener;
+			this.filter = filter;
+		}
+
+		@Override
+		public CompletableFuture<Void> onMessage(Message<T> message) {
+			if (filter.process(message)) {
+				return execute(() -> this.filteredMessageListener.onMessage(message));
+			}
+			else {
+				return CompletableFuture.completedFuture(null);
+			}
+		}
+
+		@Override
+		public CompletableFuture<Void> onMessage(Collection<Message<T>> messages) {
+			Collection<Message<T>> filteredMessages = messages.stream()
+															  .filter(filter::process)
```

Comment:
Neat pick again but please switch to for. Since size will be always maximum 10 lets switch to for loop to take benefit of performance.